### ISSUE TITLE
Correct implementation of this model?

### ISSUE THREAD

**Issue by mjack3** [VLL-HD/FrEIA#113]
Hello community. 

I am working on implementing a paper which uses this framework. As I 'm new in this tool I'd like to ask to advanced people if am I implementing correctly this architecture.

![image](https://user-images.githubusercontent.com/37028633/157633813-14ff0d17-3b00-487e-ad25-440f8d3a0019.png)

Firstly, I tried this code:

```
dims = (256, 64, 64)

inn = Ff.SequenceINN(*dims)
inn.append(Fm.AllInOneBlock, subnet_constructor=subnet_conv_3x3, permute_soft=False)
inn.append(Fm.AllInOneBlock, subnet_constructor=subnet_conv_1x1, permute_soft=False)
```

But i think the `AllInOneBlock` does not apply the permutation and the ActNorm in the order that the image shows. So I tried the next code

```
dims = (256, 64, 64)

inn = Ff.SequenceINN(*dims)
inn.append(Fm.PermuteRandom)
inn.append(Fm.ActNorm)
inn.append(Fm.GLOWCouplingBlock, subnet_constructor=subnet_conv_3x3)
inn.append(Fm.GLOWCouplingBlock, subnet_constructor=subnet_conv_1x1)
```

Which one is correct? Moreover, I am getting NaN vector when I'm adding the ``Fm.ActNorm`
`

**Comment by maaft** [comment]
I think you can get rid of NaN by:
- decreasing clamp_alpha (try 0.1 for example)
- scale incoming features

**Comment by mjack3** [comment]
@maaft Thanks I'll leave to know

**Comment by mjack3** [comment]
@maaft I am still getting NaN values in this way

```
inn = Ff.SequenceINN(*dims)
inn.append(Fm.PermuteRandom)
inn.append(Fm.ActNorm)
inn.append(Fm.GLOWCouplingBlock, subnet_constructor=subnet_conv_3x3, clamp = 1.)
```
But NaN dissapear if i put the ActNorm after the affine block:

```
inn = Ff.SequenceINN(*dims)
inn.append(Fm.PermuteRandom)
inn.append(Fm.GLOWCouplingBlock, subnet_constructor=subnet_conv_3x3, clamp = 1.)
inn.append(Fm.ActNorm)
```

Actually I dont know why it happens

**Comment by zhyzhyzhy123** [comment]
Hi

> @maaft I am still getting NaN values in this way
> 
> ```
> inn = Ff.SequenceINN(*dims)
> inn.append(Fm.PermuteRandom)
> inn.append(Fm.ActNorm)
> inn.append(Fm.GLOWCouplingBlock, subnet_constructor=subnet_conv_3x3, clamp = 1.)
> ```
> 
> But NaN dissapear if i put the ActNorm after the affine block:
> 
> ```
> inn = Ff.SequenceINN(*dims)
> inn.append(Fm.PermuteRandom)
> inn.append(Fm.GLOWCouplingBlock, subnet_constructor=subnet_conv_3x3, clamp = 1.)
> inn.append(Fm.ActNorm)
> ```
> 
> Actually I dont know why it happens

Hello there, did you fix the issue? I met the same issue but I do not know how to fix is XD

**Comment by psorrenson** [comment]
Yes, sorry about that! The maintenance of the repo has been passed on from the original maintainers. But there was a long period where it was not being maintained. Hopefully we will be a bit faster at answering issues from now on ;)

**Comment by mjack3** [comment]
You will a great job :)
